{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "15b1b873-f1a6-42f7-8d7d-9ee1d6374138",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from mylib import vn_genome_ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "176c1b8d-5ed8-404b-b5a8-ebc85223849a",
   "metadata": {},
   "outputs": [],
   "source": [
    "infile = \"data/all.csv\"\n",
    "df = pd.read_csv(infile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "119cb94b-edbb-422f-96ee-5c5e657f4775",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>HG01595</th>\n",
       "      <th>HG01596</th>\n",
       "      <th>HG01597</th>\n",
       "      <th>...</th>\n",
       "      <th>INFO:AF</th>\n",
       "      <th>INFO:EAS_AF</th>\n",
       "      <th>INFO:EUR_AF</th>\n",
       "      <th>INFO:AFR_AF</th>\n",
       "      <th>INFO:AMR_AF</th>\n",
       "      <th>INFO:SAS_AF</th>\n",
       "      <th>INFO:VT</th>\n",
       "      <th>INFO:NS</th>\n",
       "      <th>INFO:EX_TARGET</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>201700847</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=521;AN=5096;DP=23700;AF=0.1;EAS_AF=0.23;EUR...</td>\n",
       "      <td>GT</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.12</td>\n",
       "      <td>SNP</td>\n",
       "      <td>2548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>201700958</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=454;AN=5096;DP=22908;AF=0.09;EAS_AF=0.23;EU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.12</td>\n",
       "      <td>SNP</td>\n",
       "      <td>2548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>201701405</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=2;AN=5096;DP=24312;AF=0;EAS_AF=0;EUR_AF=0;A...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SNP</td>\n",
       "      <td>2548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>201701867</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=2;AN=5096;DP=233719;AF=0;EAS_AF=0;EUR_AF=0;...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SNP</td>\n",
       "      <td>2548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>201703895</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=74;AN=5096;DP=22000;AF=0.01;EAS_AF=0.02;EUR...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SNP</td>\n",
       "      <td>2548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 127 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #CHROM        POS REF ALT FILTER  \\\n",
       "0       2  201700847   C   T   PASS   \n",
       "1       2  201700958   G   A   PASS   \n",
       "2       2  201701405   G   A   PASS   \n",
       "3       2  201701867   C   T   PASS   \n",
       "4       2  201703895   G   A   PASS   \n",
       "\n",
       "                                                INFO FORMAT  HG01595  HG01596  \\\n",
       "0  AC=521;AN=5096;DP=23700;AF=0.1;EAS_AF=0.23;EUR...     GT        1        0   \n",
       "1  AC=454;AN=5096;DP=22908;AF=0.09;EAS_AF=0.23;EU...     GT        1        0   \n",
       "2  AC=2;AN=5096;DP=24312;AF=0;EAS_AF=0;EUR_AF=0;A...     GT        0        0   \n",
       "3  AC=2;AN=5096;DP=233719;AF=0;EAS_AF=0;EUR_AF=0;...     GT        0        0   \n",
       "4  AC=74;AN=5096;DP=22000;AF=0.01;EAS_AF=0.02;EUR...     GT        0        0   \n",
       "\n",
       "   HG01597  ...  INFO:AF  INFO:EAS_AF  INFO:EUR_AF  INFO:AFR_AF  INFO:AMR_AF  \\\n",
       "0        0  ...     0.10         0.23         0.07         0.04         0.04   \n",
       "1        0  ...     0.09         0.23         0.05         0.03         0.02   \n",
       "2        0  ...     0.00         0.00         0.00         0.00         0.00   \n",
       "3        0  ...     0.00         0.00         0.00         0.00         0.00   \n",
       "4        1  ...     0.01         0.02         0.00         0.04         0.00   \n",
       "\n",
       "   INFO:SAS_AF  INFO:VT  INFO:NS  INFO:EX_TARGET  gene  \n",
       "0         0.12      SNP     2548             NaN  ALS2  \n",
       "1         0.12      SNP     2548             NaN  ALS2  \n",
       "2         0.00      SNP     2548             NaN  ALS2  \n",
       "3         0.00      SNP     2548             NaN  ALS2  \n",
       "4         0.00      SNP     2548             NaN  ALS2  \n",
       "\n",
       "[5 rows x 127 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff580822-e613-4f82-9cf1-34584dec4353",
   "metadata": {},
   "source": [
    "# Filter only rows that belong to ERBB4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "64f4ecd4-bdd0-43d3-9a7f-44b3f3697750",
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "edf = df[df['gene'] == \"ERBB4\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9cdde187-0ec1-4e48-80b1-662c5af88909",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4285, 127)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fbf432c0-79d3-4438-a9fe-33056f9c6f25",
   "metadata": {},
   "outputs": [],
   "source": [
    "edf = edf[vn_genome_ids]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "65c3fa40-d3f7-4863-9fab-a1a93fdf22f2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4285, 99)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3ce70b57-17b8-4390-9dd7-e3fd692d9eb8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HG02513</th>\n",
       "      <th>HG02138</th>\n",
       "      <th>HG02075</th>\n",
       "      <th>HG02031</th>\n",
       "      <th>HG02084</th>\n",
       "      <th>HG02047</th>\n",
       "      <th>HG01597</th>\n",
       "      <th>HG01849</th>\n",
       "      <th>HG02023</th>\n",
       "      <th>HG01846</th>\n",
       "      <th>...</th>\n",
       "      <th>HG02142</th>\n",
       "      <th>HG01863</th>\n",
       "      <th>HG02026</th>\n",
       "      <th>HG01871</th>\n",
       "      <th>HG02141</th>\n",
       "      <th>HG01873</th>\n",
       "      <th>HG02049</th>\n",
       "      <th>HG02086</th>\n",
       "      <th>HG02113</th>\n",
       "      <th>HG01847</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1334</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1335</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1336</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1337</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1338</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 99 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      HG02513  HG02138  HG02075  HG02031  HG02084  HG02047  HG01597  HG01849  \\\n",
       "1334        0        0        0        0        0        0        0        0   \n",
       "1335        0        0        0        0        0        0        0        0   \n",
       "1336        0        0        0        0        0        0        0        0   \n",
       "1337        0        0        0        0        0        0        0        0   \n",
       "1338        0        0        0        0        0        0        0        0   \n",
       "\n",
       "      HG02023  HG01846  ...  HG02142  HG01863  HG02026  HG01871  HG02141  \\\n",
       "1334        0        0  ...        0        0        0        0        0   \n",
       "1335        0        0  ...        0        0        0        0        0   \n",
       "1336        0        0  ...        0        0        0        0        0   \n",
       "1337        0        0  ...        0        0        0        0        0   \n",
       "1338        0        0  ...        0        0        0        0        0   \n",
       "\n",
       "      HG01873  HG02049  HG02086  HG02113  HG01847  \n",
       "1334        0        0        0        0        0  \n",
       "1335        0        0        0        0        1  \n",
       "1336        0        0        0        0        0  \n",
       "1337        0        0        0        0        0  \n",
       "1338        0        0        0        0        0  \n",
       "\n",
       "[5 rows x 99 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ad3cfd17-903d-40ab-9a90-03283a584263",
   "metadata": {},
   "outputs": [],
   "source": [
    "sdf = pd.DataFrame(edf.sum(axis=0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "6bb28152-217d-4de1-9bed-0d10b7b8d865",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HG02513</th>\n",
       "      <td>410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG02138</th>\n",
       "      <td>362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG02075</th>\n",
       "      <td>770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG02031</th>\n",
       "      <td>489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG02084</th>\n",
       "      <td>224</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           0\n",
       "HG02513  410\n",
       "HG02138  362\n",
       "HG02075  770\n",
       "HG02031  489\n",
       "HG02084  224"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f3d6a9f4-d591-41fb-8bbd-e730437c11ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "sdf.reset_index(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "661048dd-c689-4999-af92-7e1005fd9a9e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG02513</td>\n",
       "      <td>410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG02138</td>\n",
       "      <td>362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG02075</td>\n",
       "      <td>770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG02031</td>\n",
       "      <td>489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG02084</td>\n",
       "      <td>224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>HG01873</td>\n",
       "      <td>224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>HG02049</td>\n",
       "      <td>404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>HG02086</td>\n",
       "      <td>269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>HG02113</td>\n",
       "      <td>173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>HG01847</td>\n",
       "      <td>204</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      index    0\n",
       "0   HG02513  410\n",
       "1   HG02138  362\n",
       "2   HG02075  770\n",
       "3   HG02031  489\n",
       "4   HG02084  224\n",
       "..      ...  ...\n",
       "94  HG01873  224\n",
       "95  HG02049  404\n",
       "96  HG02086  269\n",
       "97  HG02113  173\n",
       "98  HG01847  204\n",
       "\n",
       "[99 rows x 2 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "bdc900f1-8c97-4579-91a3-a370d72297ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "sdf.rename(columns={\"index\":\"genome\",0:\"count\"},inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "d84b4b8c-07a6-44fe-8cf1-f917cf4d631b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG02513</td>\n",
       "      <td>410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG02138</td>\n",
       "      <td>362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG02075</td>\n",
       "      <td>770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG02031</td>\n",
       "      <td>489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG02084</td>\n",
       "      <td>224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>HG01873</td>\n",
       "      <td>224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>HG02049</td>\n",
       "      <td>404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>HG02086</td>\n",
       "      <td>269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>HG02113</td>\n",
       "      <td>173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>HG01847</td>\n",
       "      <td>204</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     genome  count\n",
       "0   HG02513    410\n",
       "1   HG02138    362\n",
       "2   HG02075    770\n",
       "3   HG02031    489\n",
       "4   HG02084    224\n",
       "..      ...    ...\n",
       "94  HG01873    224\n",
       "95  HG02049    404\n",
       "96  HG02086    269\n",
       "97  HG02113    173\n",
       "98  HG01847    204\n",
       "\n",
       "[99 rows x 2 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "bc1d50a4-7c76-4a8e-ae4f-bfbc732577d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "cdf = sdf[sdf['count']>0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "e2605357-0028-4b52-b6ea-4ee349a69f90",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG02513</td>\n",
       "      <td>410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG02138</td>\n",
       "      <td>362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG02075</td>\n",
       "      <td>770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG02031</td>\n",
       "      <td>489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG02084</td>\n",
       "      <td>224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>HG01873</td>\n",
       "      <td>224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>HG02049</td>\n",
       "      <td>404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>HG02086</td>\n",
       "      <td>269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>HG02113</td>\n",
       "      <td>173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>HG01847</td>\n",
       "      <td>204</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     genome  count\n",
       "0   HG02513    410\n",
       "1   HG02138    362\n",
       "2   HG02075    770\n",
       "3   HG02031    489\n",
       "4   HG02084    224\n",
       "..      ...    ...\n",
       "94  HG01873    224\n",
       "95  HG02049    404\n",
       "96  HG02086    269\n",
       "97  HG02113    173\n",
       "98  HG01847    204\n",
       "\n",
       "[99 rows x 2 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cdf"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33a6fe4e-6241-4d64-bb51-d90bfc2a10c6",
   "metadata": {},
   "source": [
    "#  How many people carry variants in ERBB4? "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "749a2282-782a-4ff8-887a-d243a5e7682e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(99, 2)"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cdf.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1e51e793-6c29-420e-97d8-25efcce098d2",
   "metadata": {},
   "source": [
    "# For those people with >1000 variants, what are the genes that contribute the most variants? "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "76343c23-f93b-4bb8-89b2-c04a7fae23d7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome</th>\n",
       "      <th>variant_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01842</td>\n",
       "      <td>1339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG01872</td>\n",
       "      <td>1285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG02087</td>\n",
       "      <td>1281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG01858</td>\n",
       "      <td>1277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG01864</td>\n",
       "      <td>1269</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    genome  variant_count\n",
       "0  HG01842           1339\n",
       "1  HG01872           1285\n",
       "2  HG02087           1281\n",
       "3  HG01858           1277\n",
       "4  HG01864           1269"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "infile = \"20250801/variants_count_per_genome.csv\"\n",
    "df = pd.read_csv(infile)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "8791c298-a62e-4acf-948d-a8bd2ffb2a18",
   "metadata": {},
   "outputs": [],
   "source": [
    "gdf = df[df['variant_count']>1000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "70046fc5-f705-4c36-bc78-15dc40ac8d82",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(29, 2)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gdf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "ae5edee9-28f9-4b99-843d-f78eaff1894e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome</th>\n",
       "      <th>variant_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01842</td>\n",
       "      <td>1339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG01872</td>\n",
       "      <td>1285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG02087</td>\n",
       "      <td>1281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG01858</td>\n",
       "      <td>1277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG01864</td>\n",
       "      <td>1269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HG02048</td>\n",
       "      <td>1258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HG02127</td>\n",
       "      <td>1249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HG02025</td>\n",
       "      <td>1241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HG02064</td>\n",
       "      <td>1226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HG02137</td>\n",
       "      <td>1225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HG02131</td>\n",
       "      <td>1212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HG02116</td>\n",
       "      <td>1200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HG02078</td>\n",
       "      <td>1198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HG02067</td>\n",
       "      <td>1157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HG01845</td>\n",
       "      <td>1155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HG01869</td>\n",
       "      <td>1150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>HG02075</td>\n",
       "      <td>1138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HG01852</td>\n",
       "      <td>1117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>HG01859</td>\n",
       "      <td>1089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HG02019</td>\n",
       "      <td>1076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HG01867</td>\n",
       "      <td>1074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>HG02031</td>\n",
       "      <td>1064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HG02016</td>\n",
       "      <td>1045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>HG01862</td>\n",
       "      <td>1041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>HG02121</td>\n",
       "      <td>1031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>HG02134</td>\n",
       "      <td>1027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HG01878</td>\n",
       "      <td>1007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>HG01853</td>\n",
       "      <td>1005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>HG02023</td>\n",
       "      <td>1004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     genome  variant_count\n",
       "0   HG01842           1339\n",
       "1   HG01872           1285\n",
       "2   HG02087           1281\n",
       "3   HG01858           1277\n",
       "4   HG01864           1269\n",
       "5   HG02048           1258\n",
       "6   HG02127           1249\n",
       "7   HG02025           1241\n",
       "8   HG02064           1226\n",
       "9   HG02137           1225\n",
       "10  HG02131           1212\n",
       "11  HG02116           1200\n",
       "12  HG02078           1198\n",
       "13  HG02067           1157\n",
       "14  HG01845           1155\n",
       "15  HG01869           1150\n",
       "16  HG02075           1138\n",
       "17  HG01852           1117\n",
       "18  HG01859           1089\n",
       "19  HG02019           1076\n",
       "20  HG01867           1074\n",
       "21  HG02031           1064\n",
       "22  HG02016           1045\n",
       "23  HG01862           1041\n",
       "24  HG02121           1031\n",
       "25  HG02134           1027\n",
       "26  HG01878           1007\n",
       "27  HG01853           1005\n",
       "28  HG02023           1004"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "1b9539cd-90db-43c1-b46e-91ff4c6a1fdc",
   "metadata": {},
   "outputs": [],
   "source": [
    "top_count_genomes = list(gdf['genome'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3a6e7dd6-0bd1-44fa-8ce5-39e390dd751d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['HG01842',\n",
       " 'HG01872',\n",
       " 'HG02087',\n",
       " 'HG01858',\n",
       " 'HG01864',\n",
       " 'HG02048',\n",
       " 'HG02127',\n",
       " 'HG02025',\n",
       " 'HG02064',\n",
       " 'HG02137',\n",
       " 'HG02131',\n",
       " 'HG02116',\n",
       " 'HG02078',\n",
       " 'HG02067',\n",
       " 'HG01845',\n",
       " 'HG01869',\n",
       " 'HG02075',\n",
       " 'HG01852',\n",
       " 'HG01859',\n",
       " 'HG02019',\n",
       " 'HG01867',\n",
       " 'HG02031',\n",
       " 'HG02016',\n",
       " 'HG01862',\n",
       " 'HG02121',\n",
       " 'HG02134',\n",
       " 'HG01878',\n",
       " 'HG01853',\n",
       " 'HG02023']"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top_count_genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "b246d48e-44cb-4980-a47a-dcba564a5438",
   "metadata": {},
   "outputs": [],
   "source": [
    "infile = \"data/all.csv\"\n",
    "df = pd.read_csv(infile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "a7d1c19d-549e-4e58-8f4c-aa6e2d74d85d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(8502, 127)"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "e41e503b-f726-47d8-91a3-ae732a2139d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[top_count_genomes + ['gene']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "4fce1de0-9b6f-4a08-91c8-479a9e73a1d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(8502, 30)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "d1da212b-3d93-47c3-8d99-fa51e27e520f",
   "metadata": {},
   "outputs": [],
   "source": [
    "sdf=df.groupby([\"gene\"]).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "56f76765-f3b7-4382-aa26-4ab9e5aeb2b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HG01842</th>\n",
       "      <th>HG01872</th>\n",
       "      <th>HG02087</th>\n",
       "      <th>HG01858</th>\n",
       "      <th>HG01864</th>\n",
       "      <th>HG02048</th>\n",
       "      <th>HG02127</th>\n",
       "      <th>HG02025</th>\n",
       "      <th>HG02064</th>\n",
       "      <th>HG02137</th>\n",
       "      <th>...</th>\n",
       "      <th>HG02019</th>\n",
       "      <th>HG01867</th>\n",
       "      <th>HG02031</th>\n",
       "      <th>HG02016</th>\n",
       "      <th>HG01862</th>\n",
       "      <th>HG02121</th>\n",
       "      <th>HG02134</th>\n",
       "      <th>HG01878</th>\n",
       "      <th>HG01853</th>\n",
       "      <th>HG02023</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ALS2</th>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>46</td>\n",
       "      <td>22</td>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANG</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATXN2</th>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C9ORF72</th>\n",
       "      <td>32</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>35</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>34</td>\n",
       "      <td>24</td>\n",
       "      <td>17</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHCHD10</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         HG01842  HG01872  HG02087  HG01858  HG01864  HG02048  HG02127  \\\n",
       "gene                                                                     \n",
       "ALS2           6        7        9       23       22        9        4   \n",
       "ANG            1        2        5        4        0        0        2   \n",
       "ATXN2          3        2        7        2        2       11        2   \n",
       "C9ORF72       32        1       23        2        1        2        6   \n",
       "CHCHD10        0        0        0        1        1        0        0   \n",
       "\n",
       "         HG02025  HG02064  HG02137  ...  HG02019  HG01867  HG02031  HG02016  \\\n",
       "gene                                ...                                       \n",
       "ALS2           3        3       21  ...       21        7       44       22   \n",
       "ANG            6        1        1  ...        0        3        5        0   \n",
       "ATXN2         18        8       20  ...        1        2        3        6   \n",
       "C9ORF72       35       14       13  ...       12        0       32       34   \n",
       "CHCHD10        0        1        1  ...        0        0        0        0   \n",
       "\n",
       "         HG01862  HG02121  HG02134  HG01878  HG01853  HG02023  \n",
       "gene                                                           \n",
       "ALS2          46       22       26       24        9        3  \n",
       "ANG            0        0        2        2        1        0  \n",
       "ATXN2         16        8        7        4        3        1  \n",
       "C9ORF72       24       17       10        2       24       24  \n",
       "CHCHD10        1        0        0        0        2        2  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "16fbe7eb-222a-41fa-943f-78ba89bb88f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HG01842</th>\n",
       "      <th>HG01872</th>\n",
       "      <th>HG02087</th>\n",
       "      <th>HG01858</th>\n",
       "      <th>HG01864</th>\n",
       "      <th>HG02048</th>\n",
       "      <th>HG02127</th>\n",
       "      <th>HG02025</th>\n",
       "      <th>HG02064</th>\n",
       "      <th>HG02137</th>\n",
       "      <th>...</th>\n",
       "      <th>HG02019</th>\n",
       "      <th>HG01867</th>\n",
       "      <th>HG02031</th>\n",
       "      <th>HG02016</th>\n",
       "      <th>HG01862</th>\n",
       "      <th>HG02121</th>\n",
       "      <th>HG02134</th>\n",
       "      <th>HG01878</th>\n",
       "      <th>HG01853</th>\n",
       "      <th>HG02023</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ALS2</th>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>46</td>\n",
       "      <td>22</td>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANG</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATXN2</th>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C9ORF72</th>\n",
       "      <td>32</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>35</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>34</td>\n",
       "      <td>24</td>\n",
       "      <td>17</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHCHD10</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHMP2B</th>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHRNA3</th>\n",
       "      <td>34</td>\n",
       "      <td>36</td>\n",
       "      <td>23</td>\n",
       "      <td>18</td>\n",
       "      <td>19</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>21</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>18</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>20</td>\n",
       "      <td>32</td>\n",
       "      <td>37</td>\n",
       "      <td>25</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CREST</th>\n",
       "      <td>1</td>\n",
       "      <td>85</td>\n",
       "      <td>90</td>\n",
       "      <td>3</td>\n",
       "      <td>103</td>\n",
       "      <td>89</td>\n",
       "      <td>86</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>85</td>\n",
       "      <td>88</td>\n",
       "      <td>72</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DAO</th>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DCTN1</th>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ELP3</th>\n",
       "      <td>33</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>15</td>\n",
       "      <td>18</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>24</td>\n",
       "      <td>37</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>26</td>\n",
       "      <td>12</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>35</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERBB4</th>\n",
       "      <td>725</td>\n",
       "      <td>578</td>\n",
       "      <td>688</td>\n",
       "      <td>763</td>\n",
       "      <td>687</td>\n",
       "      <td>667</td>\n",
       "      <td>823</td>\n",
       "      <td>739</td>\n",
       "      <td>737</td>\n",
       "      <td>689</td>\n",
       "      <td>...</td>\n",
       "      <td>684</td>\n",
       "      <td>389</td>\n",
       "      <td>489</td>\n",
       "      <td>638</td>\n",
       "      <td>546</td>\n",
       "      <td>381</td>\n",
       "      <td>412</td>\n",
       "      <td>626</td>\n",
       "      <td>476</td>\n",
       "      <td>444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1</th>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIG4</th>\n",
       "      <td>84</td>\n",
       "      <td>73</td>\n",
       "      <td>34</td>\n",
       "      <td>39</td>\n",
       "      <td>60</td>\n",
       "      <td>83</td>\n",
       "      <td>21</td>\n",
       "      <td>27</td>\n",
       "      <td>60</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>62</td>\n",
       "      <td>88</td>\n",
       "      <td>72</td>\n",
       "      <td>42</td>\n",
       "      <td>39</td>\n",
       "      <td>62</td>\n",
       "      <td>48</td>\n",
       "      <td>7</td>\n",
       "      <td>79</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FUS</th>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MATR3</th>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>29</td>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>27</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>13</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEFH</th>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEK1</th>\n",
       "      <td>234</td>\n",
       "      <td>193</td>\n",
       "      <td>211</td>\n",
       "      <td>188</td>\n",
       "      <td>190</td>\n",
       "      <td>193</td>\n",
       "      <td>8</td>\n",
       "      <td>189</td>\n",
       "      <td>4</td>\n",
       "      <td>187</td>\n",
       "      <td>...</td>\n",
       "      <td>40</td>\n",
       "      <td>188</td>\n",
       "      <td>219</td>\n",
       "      <td>38</td>\n",
       "      <td>6</td>\n",
       "      <td>106</td>\n",
       "      <td>210</td>\n",
       "      <td>10</td>\n",
       "      <td>183</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OPTN</th>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>15</td>\n",
       "      <td>13</td>\n",
       "      <td>20</td>\n",
       "      <td>30</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>26</td>\n",
       "      <td>17</td>\n",
       "      <td>11</td>\n",
       "      <td>25</td>\n",
       "      <td>26</td>\n",
       "      <td>42</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PFN1</th>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PON1-3</th>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRPH</th>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SETX</th>\n",
       "      <td>53</td>\n",
       "      <td>30</td>\n",
       "      <td>19</td>\n",
       "      <td>50</td>\n",
       "      <td>39</td>\n",
       "      <td>63</td>\n",
       "      <td>40</td>\n",
       "      <td>52</td>\n",
       "      <td>27</td>\n",
       "      <td>37</td>\n",
       "      <td>...</td>\n",
       "      <td>41</td>\n",
       "      <td>87</td>\n",
       "      <td>32</td>\n",
       "      <td>44</td>\n",
       "      <td>55</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>55</td>\n",
       "      <td>27</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SIGMAR1</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOD1</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPG11</th>\n",
       "      <td>2</td>\n",
       "      <td>94</td>\n",
       "      <td>50</td>\n",
       "      <td>57</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>89</td>\n",
       "      <td>6</td>\n",
       "      <td>91</td>\n",
       "      <td>60</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>95</td>\n",
       "      <td>3</td>\n",
       "      <td>50</td>\n",
       "      <td>91</td>\n",
       "      <td>124</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SQSTM1</th>\n",
       "      <td>17</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TAF15</th>\n",
       "      <td>25</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>24</td>\n",
       "      <td>29</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>27</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>27</td>\n",
       "      <td>25</td>\n",
       "      <td>24</td>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TARDBP</th>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TUBA4A</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VAPB</th>\n",
       "      <td>25</td>\n",
       "      <td>23</td>\n",
       "      <td>24</td>\n",
       "      <td>22</td>\n",
       "      <td>17</td>\n",
       "      <td>37</td>\n",
       "      <td>5</td>\n",
       "      <td>38</td>\n",
       "      <td>19</td>\n",
       "      <td>39</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>36</td>\n",
       "      <td>20</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VCP</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hnRNPA2B1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hnRNPa1</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>34 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           HG01842  HG01872  HG02087  HG01858  HG01864  HG02048  HG02127  \\\n",
       "gene                                                                       \n",
       "ALS2             6        7        9       23       22        9        4   \n",
       "ANG              1        2        5        4        0        0        2   \n",
       "ATXN2            3        2        7        2        2       11        2   \n",
       "C9ORF72         32        1       23        2        1        2        6   \n",
       "CHCHD10          0        0        0        1        1        0        0   \n",
       "CHMP2B           7        3        7       15       11        8       13   \n",
       "CHRNA3          34       36       23       18       19        6       20   \n",
       "CREST            1       85       90        3      103       89       86   \n",
       "DAO              1       11        3        1        2        2        3   \n",
       "DCTN1            3        5        2        3        3        6        2   \n",
       "ELP3            33       15       17       15       18        5       13   \n",
       "ERBB4          725      578      688      763      687      667      823   \n",
       "EWSR1            9        8        7        5       11       10        8   \n",
       "FIG4            84       73       34       39       60       83       21   \n",
       "FUS              4        5        4        3        4        0        0   \n",
       "MATR3            0       29        0        2        5        3       29   \n",
       "NEFH             9        2        4        2        1        2        2   \n",
       "NEK1           234      193      211      188      190      193        8   \n",
       "OPTN            13       28       21       14       12       15       13   \n",
       "PFN1             4        0        2        4        4        0        4   \n",
       "PON1-3           2       21        9        1        5        1        2   \n",
       "PRPH             0        6        0        0        0        0        0   \n",
       "SETX            53       30       19       50       39       63       40   \n",
       "SIGMAR1          3        0        0        1        0        1        0   \n",
       "SOD1             0        2        0        1        0        3        0   \n",
       "SPG11            2       94       50       57        5        3       89   \n",
       "SQSTM1          17       16       14        6       11        6        6   \n",
       "TAF15           25        3        1       25       25       24       29   \n",
       "TARDBP           5        6        0        0        4        5        3   \n",
       "TUBA4A           0        0        0        0        0        0        0   \n",
       "VAPB            25       23       24       22       17       37        5   \n",
       "VCP              2        0        0        1        0        0       12   \n",
       "hnRNPA2B1        1        1        7        6        7        3        4   \n",
       "hnRNPa1          1        0        0        0        0        1        0   \n",
       "\n",
       "           HG02025  HG02064  HG02137  ...  HG02019  HG01867  HG02031  HG02016  \\\n",
       "gene                                  ...                                       \n",
       "ALS2             3        3       21  ...       21        7       44       22   \n",
       "ANG              6        1        1  ...        0        3        5        0   \n",
       "ATXN2           18        8       20  ...        1        2        3        6   \n",
       "C9ORF72         35       14       13  ...       12        0       32       34   \n",
       "CHCHD10          0        1        1  ...        0        0        0        0   \n",
       "CHMP2B          10        5        5  ...        9       10        8        5   \n",
       "CHRNA3          15       21       15  ...       25       25       18       19   \n",
       "CREST            1       87        3  ...       19       19        2        2   \n",
       "DAO             10        1        2  ...        5        1        2        1   \n",
       "DCTN1            3        4        2  ...        0        1        0        1   \n",
       "ELP3            28       24       37  ...       28       16       14       14   \n",
       "ERBB4          739      737      689  ...      684      389      489      638   \n",
       "EWSR1            4       12       10  ...       12        5        2        5   \n",
       "FIG4            27       60        4  ...       62       88       72       42   \n",
       "FUS              1        4        3  ...        4        4        0        3   \n",
       "MATR3            0       30        3  ...        4       29       27        2   \n",
       "NEFH             2       10        5  ...        3        1        2       10   \n",
       "NEK1           189        4      187  ...       40      188      219       38   \n",
       "OPTN            20       30       22  ...        6       18       12       26   \n",
       "PFN1             0        0        0  ...        0        1        4        0   \n",
       "PON1-3           8        6        6  ...       19        1        0        5   \n",
       "PRPH             0        0        0  ...        0        0        0        0   \n",
       "SETX            52       27       37  ...       41       87       32       44   \n",
       "SIGMAR1          1        1        0  ...        1        2        2        1   \n",
       "SOD1             0        1        0  ...        0        0        2        0   \n",
       "SPG11            6       91       60  ...        3       95        3       50   \n",
       "SQSTM1           6       10        4  ...       12        5        3       15   \n",
       "TAF15            6        2       27  ...       29       27       25       24   \n",
       "TARDBP           5        6        6  ...        6        6        4        6   \n",
       "TUBA4A           3        0        0  ...        0        3        0        0   \n",
       "VAPB            38       19       39  ...       25       38       16       29   \n",
       "VCP              3        1        0  ...        1        2       13        1   \n",
       "hnRNPA2B1        2        6        3  ...        0        0        9        1   \n",
       "hnRNPa1          0        0        0  ...        4        1        0        1   \n",
       "\n",
       "           HG01862  HG02121  HG02134  HG01878  HG01853  HG02023  \n",
       "gene                                                             \n",
       "ALS2            46       22       26       24        9        3  \n",
       "ANG              0        0        2        2        1        0  \n",
       "ATXN2           16        8        7        4        3        1  \n",
       "C9ORF72         24       17       10        2       24       24  \n",
       "CHCHD10          1        0        0        0        2        2  \n",
       "CHMP2B          12       13        1       15       17       13  \n",
       "CHRNA3          19       20       32       37       25       38  \n",
       "CREST            0       85       88       72        1       87  \n",
       "DAO              7        9        1       11        2        3  \n",
       "DCTN1            0        0        2        1        4        2  \n",
       "ELP3            26       12       15       15       35       53  \n",
       "ERBB4          546      381      412      626      476      444  \n",
       "EWSR1            7       11       13        8        9        7  \n",
       "FIG4            39       62       48        7       79       14  \n",
       "FUS              4        0        4        3        0        1  \n",
       "MATR3           27       28        0       28       13       27  \n",
       "NEFH             2        5        3        8        2        3  \n",
       "NEK1             6      106      210       10      183       43  \n",
       "OPTN            17       11       25       26       42       14  \n",
       "PFN1             0        0        0        4        3        4  \n",
       "PON1-3           9        6       20        9        6        3  \n",
       "PRPH             0        6        0        0        0        6  \n",
       "SETX            55       52       52       55       27       64  \n",
       "SIGMAR1          1        1        0        1        1        1  \n",
       "SOD1             0        1        0        0        0        0  \n",
       "SPG11           91      124        6        7        3       90  \n",
       "SQSTM1           7        4        8        7        5       11  \n",
       "TAF15           27        2        1        1       26       24  \n",
       "TARDBP           0        3        0        2        2        4  \n",
       "TUBA4A           4        0        3        0        0        0  \n",
       "VAPB            39       39       36       20        4       14  \n",
       "VCP              3        1        1        0        1        0  \n",
       "hnRNPA2B1        6        2        1        1        0        3  \n",
       "hnRNPa1          0        0        0        1        0        1  \n",
       "\n",
       "[34 rows x 29 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "b3ff7b7b-8657-4fe7-8d67-99612ab2a5b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "top_genes_per_genome = dict()\n",
    "num_top_genes = 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "7bc8c610-cb74-4bf8-81e4-e48e2211187f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         HG01842\n",
      "gene            \n",
      "ERBB4        725\n",
      "NEK1         234\n",
      "FIG4          84\n",
      "SETX          53\n",
      "CHRNA3        34\n",
      "ELP3          33\n",
      "C9ORF72       32\n",
      "TAF15         25\n",
      "VAPB          25\n",
      "SQSTM1        17\n"
     ]
    }
   ],
   "source": [
    "col = 'HG01842'\n",
    "top_genes = sdf[[col]].nlargest(num_top_genes,col)\n",
    "print(top_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "223b7a34-7212-4465-85c6-f530929ef1b7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         HG01842\n",
      "gene            \n",
      "ERBB4        725\n",
      "NEK1         234\n",
      "FIG4          84\n",
      "SETX          53\n",
      "CHRNA3        34\n",
      "ELP3          33\n",
      "C9ORF72       32\n",
      "TAF15         25\n",
      "VAPB          25\n",
      "SQSTM1        17\n",
      "------------------------\n",
      "        HG01872\n",
      "gene           \n",
      "ERBB4       578\n",
      "NEK1        193\n",
      "SPG11        94\n",
      "CREST        85\n",
      "FIG4         73\n",
      "CHRNA3       36\n",
      "SETX         30\n",
      "MATR3        29\n",
      "OPTN         28\n",
      "VAPB         23\n",
      "------------------------\n",
      "         HG02087\n",
      "gene            \n",
      "ERBB4        688\n",
      "NEK1         211\n",
      "CREST         90\n",
      "SPG11         50\n",
      "FIG4          34\n",
      "VAPB          24\n",
      "C9ORF72       23\n",
      "CHRNA3        23\n",
      "OPTN          21\n",
      "SETX          19\n",
      "------------------------\n",
      "        HG01858\n",
      "gene           \n",
      "ERBB4       763\n",
      "NEK1        188\n",
      "SPG11        57\n",
      "SETX         50\n",
      "FIG4         39\n",
      "TAF15        25\n",
      "ALS2         23\n",
      "VAPB         22\n",
      "CHRNA3       18\n",
      "CHMP2B       15\n",
      "------------------------\n",
      "        HG01864\n",
      "gene           \n",
      "ERBB4       687\n",
      "NEK1        190\n",
      "CREST       103\n",
      "FIG4         60\n",
      "SETX         39\n",
      "TAF15        25\n",
      "ALS2         22\n",
      "CHRNA3       19\n",
      "ELP3         18\n",
      "VAPB         17\n",
      "------------------------\n",
      "       HG02048\n",
      "gene          \n",
      "ERBB4      667\n",
      "NEK1       193\n",
      "CREST       89\n",
      "FIG4        83\n",
      "SETX        63\n",
      "VAPB        37\n",
      "TAF15       24\n",
      "OPTN        15\n",
      "ATXN2       11\n",
      "EWSR1       10\n",
      "------------------------\n",
      "        HG02127\n",
      "gene           \n",
      "ERBB4       823\n",
      "SPG11        89\n",
      "CREST        86\n",
      "SETX         40\n",
      "MATR3        29\n",
      "TAF15        29\n",
      "FIG4         21\n",
      "CHRNA3       20\n",
      "CHMP2B       13\n",
      "ELP3         13\n",
      "------------------------\n",
      "         HG02025\n",
      "gene            \n",
      "ERBB4        739\n",
      "NEK1         189\n",
      "SETX          52\n",
      "VAPB          38\n",
      "C9ORF72       35\n",
      "ELP3          28\n",
      "FIG4          27\n",
      "OPTN          20\n",
      "ATXN2         18\n",
      "CHRNA3        15\n",
      "------------------------\n",
      "        HG02064\n",
      "gene           \n",
      "ERBB4       737\n",
      "SPG11        91\n",
      "CREST        87\n",
      "FIG4         60\n",
      "MATR3        30\n",
      "OPTN         30\n",
      "SETX         27\n",
      "ELP3         24\n",
      "CHRNA3       21\n",
      "VAPB         19\n",
      "------------------------\n",
      "       HG02137\n",
      "gene          \n",
      "ERBB4      689\n",
      "NEK1       187\n",
      "SPG11       60\n",
      "VAPB        39\n",
      "ELP3        37\n",
      "SETX        37\n",
      "TAF15       27\n",
      "OPTN        22\n",
      "ALS2        21\n",
      "ATXN2       20\n",
      "------------------------\n",
      "         HG02131\n",
      "gene            \n",
      "ERBB4        658\n",
      "NEK1         188\n",
      "FIG4          58\n",
      "SETX          49\n",
      "VAPB          36\n",
      "ELP3          34\n",
      "OPTN          24\n",
      "TAF15         24\n",
      "ALS2          23\n",
      "C9ORF72       22\n",
      "------------------------\n",
      "        HG02116\n",
      "gene           \n",
      "ERBB4       611\n",
      "NEK1        184\n",
      "CREST        88\n",
      "ALS2         62\n",
      "SPG11        51\n",
      "FIG4         43\n",
      "CHRNA3       23\n",
      "OPTN         21\n",
      "PON1-3       19\n",
      "VAPB         18\n",
      "------------------------\n",
      "         HG02078\n",
      "gene            \n",
      "ERBB4        687\n",
      "NEK1         233\n",
      "SPG11         58\n",
      "ELP3          43\n",
      "SETX          27\n",
      "VAPB          25\n",
      "FIG4          20\n",
      "CHRNA3        16\n",
      "C9ORF72       15\n",
      "SQSTM1        10\n",
      "------------------------\n",
      "        HG02067\n",
      "gene           \n",
      "ERBB4       482\n",
      "NEK1        190\n",
      "SPG11       117\n",
      "SETX         51\n",
      "FIG4         44\n",
      "ELP3         35\n",
      "CHRNA3       34\n",
      "MATR3        27\n",
      "TAF15        25\n",
      "ALS2         23\n",
      "------------------------\n",
      "         HG01845\n",
      "gene            \n",
      "ERBB4        871\n",
      "SETX          68\n",
      "SPG11         59\n",
      "VAPB          25\n",
      "C9ORF72       24\n",
      "ATXN2         18\n",
      "FIG4          12\n",
      "OPTN          12\n",
      "EWSR1         10\n",
      "CHMP2B         8\n",
      "------------------------\n",
      "         HG01869\n",
      "gene            \n",
      "ERBB4        609\n",
      "NEK1         186\n",
      "SETX          51\n",
      "CHRNA3        34\n",
      "FIG4          32\n",
      "OPTN          30\n",
      "MATR3         28\n",
      "ELP3          27\n",
      "C9ORF72       26\n",
      "TAF15         25\n",
      "------------------------\n",
      "         HG02075\n",
      "gene            \n",
      "ERBB4        770\n",
      "VAPB          58\n",
      "SETX          54\n",
      "MATR3         31\n",
      "CHRNA3        30\n",
      "FIG4          25\n",
      "OPTN          25\n",
      "SQSTM1        16\n",
      "C9ORF72       15\n",
      "ELP3          14\n",
      "------------------------\n",
      "        HG01852\n",
      "gene           \n",
      "ERBB4       493\n",
      "NEK1        209\n",
      "CREST        86\n",
      "FIG4         56\n",
      "SPG11        56\n",
      "MATR3        29\n",
      "SETX         25\n",
      "OPTN         20\n",
      "PON1-3       19\n",
      "ELP3         15\n",
      "------------------------\n",
      "        HG01859\n",
      "gene           \n",
      "ERBB4       438\n",
      "NEK1        188\n",
      "SPG11        92\n",
      "CREST        86\n",
      "FIG4         54\n",
      "SETX         46\n",
      "ALS2         26\n",
      "VAPB         25\n",
      "PON1-3       21\n",
      "CHRNA3       18\n",
      "------------------------\n",
      "        HG02019\n",
      "gene           \n",
      "ERBB4       684\n",
      "FIG4         62\n",
      "SETX         41\n",
      "NEK1         40\n",
      "TAF15        29\n",
      "ELP3         28\n",
      "CHRNA3       25\n",
      "VAPB         25\n",
      "ALS2         21\n",
      "CREST        19\n",
      "------------------------\n",
      "        HG01867\n",
      "gene           \n",
      "ERBB4       389\n",
      "NEK1        188\n",
      "SPG11        95\n",
      "FIG4         88\n",
      "SETX         87\n",
      "VAPB         38\n",
      "MATR3        29\n",
      "TAF15        27\n",
      "CHRNA3       25\n",
      "CREST        19\n",
      "------------------------\n",
      "         HG02031\n",
      "gene            \n",
      "ERBB4        489\n",
      "NEK1         219\n",
      "FIG4          72\n",
      "ALS2          44\n",
      "C9ORF72       32\n",
      "SETX          32\n",
      "MATR3         27\n",
      "TAF15         25\n",
      "CHRNA3        18\n",
      "VAPB          16\n",
      "------------------------\n",
      "         HG02016\n",
      "gene            \n",
      "ERBB4        638\n",
      "SPG11         50\n",
      "SETX          44\n",
      "FIG4          42\n",
      "NEK1          38\n",
      "C9ORF72       34\n",
      "VAPB          29\n",
      "OPTN          26\n",
      "TAF15         24\n",
      "ALS2          22\n",
      "------------------------\n",
      "         HG01862\n",
      "gene            \n",
      "ERBB4        546\n",
      "SPG11         91\n",
      "SETX          55\n",
      "ALS2          46\n",
      "FIG4          39\n",
      "VAPB          39\n",
      "MATR3         27\n",
      "TAF15         27\n",
      "ELP3          26\n",
      "C9ORF72       24\n",
      "------------------------\n",
      "        HG02121\n",
      "gene           \n",
      "ERBB4       381\n",
      "SPG11       124\n",
      "NEK1        106\n",
      "CREST        85\n",
      "FIG4         62\n",
      "SETX         52\n",
      "VAPB         39\n",
      "MATR3        28\n",
      "ALS2         22\n",
      "CHRNA3       20\n",
      "------------------------\n",
      "        HG02134\n",
      "gene           \n",
      "ERBB4       412\n",
      "NEK1        210\n",
      "CREST        88\n",
      "SETX         52\n",
      "FIG4         48\n",
      "VAPB         36\n",
      "CHRNA3       32\n",
      "ALS2         26\n",
      "OPTN         25\n",
      "PON1-3       20\n",
      "------------------------\n",
      "        HG01878\n",
      "gene           \n",
      "ERBB4       626\n",
      "CREST        72\n",
      "SETX         55\n",
      "CHRNA3       37\n",
      "MATR3        28\n",
      "OPTN         26\n",
      "ALS2         24\n",
      "VAPB         20\n",
      "CHMP2B       15\n",
      "ELP3         15\n",
      "------------------------\n",
      "         HG01853\n",
      "gene            \n",
      "ERBB4        476\n",
      "NEK1         183\n",
      "FIG4          79\n",
      "OPTN          42\n",
      "ELP3          35\n",
      "SETX          27\n",
      "TAF15         26\n",
      "CHRNA3        25\n",
      "C9ORF72       24\n",
      "CHMP2B        17\n",
      "------------------------\n",
      "         HG02023\n",
      "gene            \n",
      "ERBB4        444\n",
      "SPG11         90\n",
      "CREST         87\n",
      "SETX          64\n",
      "ELP3          53\n",
      "NEK1          43\n",
      "CHRNA3        38\n",
      "MATR3         27\n",
      "C9ORF72       24\n",
      "TAF15         24\n",
      "------------------------\n"
     ]
    }
   ],
   "source": [
    "for col in sdf.columns:\n",
    "    top_genes = sdf[[col]].nlargest(num_top_genes,col)\n",
    "    print(top_genes)\n",
    "    print(\"------------------------\")\n",
    "    #break\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1e287b9-9493-41b7-b21f-62aaa24d5784",
   "metadata": {},
   "source": [
    "## I expected a more equal distribution (flat) but this is clearly skewed towards a subset in the population. We can also display the different genes using different colours in this bar chart (i.e make it into a stacked bar chart). If you want to make a stacked bar chart, to avoid having too many samples, take a subset of samples with >1000 variants (subset 1), and a subset of samples with <600 variants (subset 2), and make the charts for these. I used 1000 and 600 arbitrarily, you can use (mean + 2SD) and (mean - 2SD) as the threshold to select your subsets.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "c7bc7fdb-504c-4e5b-a77a-0292c62caa5c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome</th>\n",
       "      <th>variant_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01842</td>\n",
       "      <td>1339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG01872</td>\n",
       "      <td>1285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG02087</td>\n",
       "      <td>1281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG01858</td>\n",
       "      <td>1277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG01864</td>\n",
       "      <td>1269</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    genome  variant_count\n",
       "0  HG01842           1339\n",
       "1  HG01872           1285\n",
       "2  HG02087           1281\n",
       "3  HG01858           1277\n",
       "4  HG01864           1269"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "infile = \"20250801/variants_count_per_genome.csv\"\n",
    "df = pd.read_csv(infile)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "359cfd76-7385-4081-b450-9f3acf9711cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.float64(886.6262626262626)"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean = df['variant_count'].mean()\n",
    "mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "cca3e3ed-3ee1-44d3-ba6f-2349e21809df",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.float64(215.18205798520154)"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "std = df['variant_count'].std()\n",
    "std"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "7ed1c9e0-07dd-4bf6-a4e0-7e3481e58532",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(np.float64(1316.9903785966658), np.float64(456.2621466558595))"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean + 2*std, mean - 2*std"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d7045dc7-27ba-48f5-8e24-65a0f5d5380a",
   "metadata": {},
   "source": [
    "## To look at how many genomes/individuals/people share the same variants, remember that we have the columns with a list of genome names (that we talked about removing because they are too long). We might want to see if there is an overlap in sample names anywhere.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "2868cb2a-4bf3-4006-b1b5-34f1b66f8c44",
   "metadata": {},
   "outputs": [],
   "source": [
    "infile = \"data/all.csv\"\n",
    "df = pd.read_csv(infile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "693f7f0c-27f5-4b3c-8543-48d24742edb1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>HG01595</th>\n",
       "      <th>HG01596</th>\n",
       "      <th>HG01597</th>\n",
       "      <th>...</th>\n",
       "      <th>INFO:AF</th>\n",
       "      <th>INFO:EAS_AF</th>\n",
       "      <th>INFO:EUR_AF</th>\n",
       "      <th>INFO:AFR_AF</th>\n",
       "      <th>INFO:AMR_AF</th>\n",
       "      <th>INFO:SAS_AF</th>\n",
       "      <th>INFO:VT</th>\n",
       "      <th>INFO:NS</th>\n",
       "      <th>INFO:EX_TARGET</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>201700847</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=521;AN=5096;DP=23700;AF=0.1;EAS_AF=0.23;EUR...</td>\n",
       "      <td>GT</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.12</td>\n",
       "      <td>SNP</td>\n",
       "      <td>2548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>201700958</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=454;AN=5096;DP=22908;AF=0.09;EAS_AF=0.23;EU...</td>\n",
       "      <td>GT</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.12</td>\n",
       "      <td>SNP</td>\n",
       "      <td>2548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>201701405</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=2;AN=5096;DP=24312;AF=0;EAS_AF=0;EUR_AF=0;A...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SNP</td>\n",
       "      <td>2548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>201701867</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=2;AN=5096;DP=233719;AF=0;EAS_AF=0;EUR_AF=0;...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SNP</td>\n",
       "      <td>2548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>201703895</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=74;AN=5096;DP=22000;AF=0.01;EAS_AF=0.02;EUR...</td>\n",
       "      <td>GT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SNP</td>\n",
       "      <td>2548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 127 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #CHROM        POS REF ALT FILTER  \\\n",
       "0       2  201700847   C   T   PASS   \n",
       "1       2  201700958   G   A   PASS   \n",
       "2       2  201701405   G   A   PASS   \n",
       "3       2  201701867   C   T   PASS   \n",
       "4       2  201703895   G   A   PASS   \n",
       "\n",
       "                                                INFO FORMAT  HG01595  HG01596  \\\n",
       "0  AC=521;AN=5096;DP=23700;AF=0.1;EAS_AF=0.23;EUR...     GT        1        0   \n",
       "1  AC=454;AN=5096;DP=22908;AF=0.09;EAS_AF=0.23;EU...     GT        1        0   \n",
       "2  AC=2;AN=5096;DP=24312;AF=0;EAS_AF=0;EUR_AF=0;A...     GT        0        0   \n",
       "3  AC=2;AN=5096;DP=233719;AF=0;EAS_AF=0;EUR_AF=0;...     GT        0        0   \n",
       "4  AC=74;AN=5096;DP=22000;AF=0.01;EAS_AF=0.02;EUR...     GT        0        0   \n",
       "\n",
       "   HG01597  ...  INFO:AF  INFO:EAS_AF  INFO:EUR_AF  INFO:AFR_AF  INFO:AMR_AF  \\\n",
       "0        0  ...     0.10         0.23         0.07         0.04         0.04   \n",
       "1        0  ...     0.09         0.23         0.05         0.03         0.02   \n",
       "2        0  ...     0.00         0.00         0.00         0.00         0.00   \n",
       "3        0  ...     0.00         0.00         0.00         0.00         0.00   \n",
       "4        1  ...     0.01         0.02         0.00         0.04         0.00   \n",
       "\n",
       "   INFO:SAS_AF  INFO:VT  INFO:NS  INFO:EX_TARGET  gene  \n",
       "0         0.12      SNP     2548             NaN  ALS2  \n",
       "1         0.12      SNP     2548             NaN  ALS2  \n",
       "2         0.00      SNP     2548             NaN  ALS2  \n",
       "3         0.00      SNP     2548             NaN  ALS2  \n",
       "4         0.00      SNP     2548             NaN  ALS2  \n",
       "\n",
       "[5 rows x 127 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "0f922092-bf0a-487e-9609-e7047aad22af",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[['POS','genome','gene']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "baf55f4f-0db3-4ff2-88d0-81bf69429603",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>genome</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>201700847</td>\n",
       "      <td>HG01595,HG01600,HG01840,HG01846,HG01848,HG0185...</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>201700958</td>\n",
       "      <td>HG01595,HG01600,HG01840,HG01846,HG01848,HG0185...</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>201701405</td>\n",
       "      <td>HG01853</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>201701867</td>\n",
       "      <td>HG02138</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>201703895</td>\n",
       "      <td>HG01597,HG01862,HG02029,HG02031,HG02116,HG0251...</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>201704663</td>\n",
       "      <td>HG02088,HG02134</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>201705906</td>\n",
       "      <td>HG01859,HG02049,HG02076,HG02121</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>201706007</td>\n",
       "      <td>HG02060</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>201706918</td>\n",
       "      <td>HG02127</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>201707434</td>\n",
       "      <td>HG01846,HG02023,HG02025,HG02028,HG02029,HG0208...</td>\n",
       "      <td>ALS2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         POS                                             genome  gene\n",
       "0  201700847  HG01595,HG01600,HG01840,HG01846,HG01848,HG0185...  ALS2\n",
       "1  201700958  HG01595,HG01600,HG01840,HG01846,HG01848,HG0185...  ALS2\n",
       "2  201701405                                            HG01853  ALS2\n",
       "3  201701867                                            HG02138  ALS2\n",
       "4  201703895  HG01597,HG01862,HG02029,HG02031,HG02116,HG0251...  ALS2\n",
       "5  201704663                                    HG02088,HG02134  ALS2\n",
       "6  201705906                    HG01859,HG02049,HG02076,HG02121  ALS2\n",
       "7  201706007                                            HG02060  ALS2\n",
       "8  201706918                                            HG02127  ALS2\n",
       "9  201707434  HG01846,HG02023,HG02025,HG02028,HG02029,HG0208...  ALS2"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "ae0183fd-0cae-45ad-a7c7-e78185a98663",
   "metadata": {},
   "outputs": [],
   "source": [
    "l1 = df.loc[0]['genome'].split(\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "2dcc58d3-35f4-4d9d-ba97-c0c668352587",
   "metadata": {},
   "outputs": [],
   "source": [
    "l2 = df.loc[4]['genome'].split(\",\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "353d4ed6-71c4-4746-89f4-08c6aa5fb4bc",
   "metadata": {},
   "source": [
    "# Find the intersection of 2 genome lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "6040666b-8907-4e89-96a7-4e3701c7fc16",
   "metadata": {},
   "outputs": [],
   "source": [
    "s= set(l1) & set (l2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "8428fafb-eda0-45eb-978a-d267ecd01093",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(list(s))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae43ca6f-4518-4013-9f7e-b2c9011e47dc",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
