{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "4770d4a5-4b8b-45ac-afc2-8333b672e161",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from scipy import stats\n",
    "import matplotlib.pyplot as plt\n",
    "from os.path import join\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e092ed32-18ac-4978-821a-2b6474612aba",
   "metadata": {},
   "source": [
    "## 1. Compare between 2 lists (populations)\n",
    "### Each list has 34 numbers (each number represents the mean amount of variant counts per gene, across all genomes)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "68e389a2-2e21-4afe-8f44-67667fd121d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "khv_genome_ids = ['HG02069', 'HG01846', 'HG02078', 'HG01851', 'HG01865', 'HG01862', 'HG02137', 'HG02050', 'HG02019', 'HG02029', 'HG02522', 'HG01850', 'HG02072', 'HG02139', 'HG02058', 'HG02081', 'HG01858', 'HG01867', 'HG02032', 'HG02122', 'HG01860', 'HG02016', 'HG01869', 'HG02061', 'HG02084', 'HG01852', 'HG02026', 'HG02521', 'HG01848', 'HG02130', 'HG02140', 'HG02113', 'HG02075', 'HG01598', 'HG02047', 'HG02121', 'HG01599', 'HG02136', 'HG01863', 'HG01874', 'HG02138', 'HG02025', 'HG01844', 'HG01845', 'HG02049', 'HG01870', 'HG02133', 'HG02088', 'HG02028', 'HG02076', 'HG01842', 'HG02023', 'HG02142', 'HG02035', 'HG02085', 'HG02086', 'HG02513', 'HG01873', 'HG02134', 'HG01871', 'HG01600', 'HG01868', 'HG02060', 'HG01878', 'HG02020', 'HG02127', 'HG01864', 'HG01595', 'HG02048', 'HG01847', 'HG01859', 'HG01861', 'HG01849', 'HG01843', 'HG02031', 'HG02512', 'HG01857', 'HG01853', 'HG02082', 'HG02073', 'HG02128', 'HG01855', 'HG02064', 'HG02040', 'HG02087', 'HG01597', 'HG02079', 'HG01872', 'HG02070', 'HG02131', 'HG02017', 'HG01866', 'HG02116', 'HG01840', 'HG01841', 'HG01596', 'HG02141', 'HG02067', 'HG02057']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "7435784f-0acb-4e42-9f59-45ded58731f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "root_dir = join(Path.home(),\"cp_als\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "11e8619e-8b20-425d-a0c7-a92f6e158492",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>HG02069</th>\n",
       "      <th>HG01846</th>\n",
       "      <th>HG02078</th>\n",
       "      <th>HG01851</th>\n",
       "      <th>HG01865</th>\n",
       "      <th>HG01862</th>\n",
       "      <th>HG02137</th>\n",
       "      <th>HG02050</th>\n",
       "      <th>HG02019</th>\n",
       "      <th>...</th>\n",
       "      <th>HG02131</th>\n",
       "      <th>HG02017</th>\n",
       "      <th>HG01866</th>\n",
       "      <th>HG02116</th>\n",
       "      <th>HG01840</th>\n",
       "      <th>HG01841</th>\n",
       "      <th>HG01596</th>\n",
       "      <th>HG02141</th>\n",
       "      <th>HG02067</th>\n",
       "      <th>HG02057</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 100 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene  HG02069  HG01846  HG02078  HG01851  HG01865  HG01862  HG02137  \\\n",
       "0  TARDBP        0        0        0        0        0        0        0   \n",
       "1  TARDBP        0        0        0        0        0        0        0   \n",
       "2  TARDBP        0        0        0        0        0        0        0   \n",
       "3  TARDBP        0        0        0        0        0        0        0   \n",
       "4  TARDBP        0        0        0        0        1        0        1   \n",
       "\n",
       "   HG02050  HG02019  ...  HG02131  HG02017  HG01866  HG02116  HG01840  \\\n",
       "0        0        0  ...        0        0        0        0        0   \n",
       "1        0        0  ...        0        0        0        0        0   \n",
       "2        0        0  ...        0        0        0        0        0   \n",
       "3        0        1  ...        0        0        0        0        0   \n",
       "4        1        1  ...        0        1        1        0        0   \n",
       "\n",
       "   HG01841  HG01596  HG02141  HG02067  HG02057  \n",
       "0        0        0        0        0        0  \n",
       "1        0        0        0        0        0  \n",
       "2        0        0        0        0        0  \n",
       "3        0        0        0        0        0  \n",
       "4        0        0        0        0        0  \n",
       "\n",
       "[5 rows x 100 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "khv_infile = join(root_dir,\"populations/output/KHV/all.csv\")\n",
    "khv_df=pd.read_csv(khv_infile)\n",
    "khv_df = khv_df[[\"gene\"]+ khv_genome_ids ]\n",
    "khv_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "330c4aaa-9a6d-414a-bdeb-efa5c7884dc2",
   "metadata": {},
   "outputs": [],
   "source": [
    "jpt_genome_ids = ['NA18999', 'NA18960', 'NA18976', 'NA19081', 'NA18987', 'NA18985', 'NA19077', 'NA19006', 'NA18948', 'NA19009', 'NA18959', 'NA19060', 'NA19091', 'NA18965', 'NA19059', 'NA19003', 'NA19000', 'NA18941', 'NA19076', 'NA19080', 'NA18994', 'NA18973', 'NA18974', 'NA18961', 'NA19063', 'NA18979', 'NA18947', 'NA19087', 'NA18981', 'NA19064', 'NA18945', 'NA19082', 'NA19070', 'NA18972', 'NA18949', 'NA18975', 'NA18989', 'NA18967', 'NA19083', 'NA18997', 'NA18966', 'NA19090', 'NA18983', 'NA18952', 'NA19062', 'NA19012', 'NA18955', 'NA18940', 'NA19004', 'NA18970', 'NA18946', 'NA18971', 'NA18988', 'NA18977', 'NA18980', 'NA18939', 'NA19088', 'NA18957', 'NA19065', 'NA19078', 'NA19074', 'NA18968', 'NA18982', 'NA19011', 'NA18995', 'NA19054', 'NA18990', 'NA18978', 'NA18943', 'NA19066', 'NA18953', 'NA19056', 'NA18962', 'NA19072', 'NA19068', 'NA18964', 'NA19067', 'NA18951', 'NA18992', 'NA19057', 'NA18963', 'NA19089', 'NA18984', 'NA18956', 'NA19002', 'NA18993', 'NA19055', 'NA18969', 'NA19005', 'NA19007', 'NA19010', 'NA19079', 'NA18991', 'NA18986', 'NA18942', 'NA19058', 'NA18944', 'NA19084', 'NA19075', 'NA18950', 'NA19086', 'NA19085', 'NA19001', 'NA18954', 'NA18998']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "5e364562-5dab-4d91-ae52-03f2bc4800e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>NA18999</th>\n",
       "      <th>NA18960</th>\n",
       "      <th>NA18976</th>\n",
       "      <th>NA19081</th>\n",
       "      <th>NA18987</th>\n",
       "      <th>NA18985</th>\n",
       "      <th>NA19077</th>\n",
       "      <th>NA19006</th>\n",
       "      <th>NA18948</th>\n",
       "      <th>...</th>\n",
       "      <th>NA19058</th>\n",
       "      <th>NA18944</th>\n",
       "      <th>NA19084</th>\n",
       "      <th>NA19075</th>\n",
       "      <th>NA18950</th>\n",
       "      <th>NA19086</th>\n",
       "      <th>NA19085</th>\n",
       "      <th>NA19001</th>\n",
       "      <th>NA18954</th>\n",
       "      <th>NA18998</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 106 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene  NA18999  NA18960  NA18976  NA19081  NA18987  NA18985  NA19077  \\\n",
       "0  TARDBP        0        0        0        0        0        1        0   \n",
       "1  TARDBP        0        0        0        0        0        1        0   \n",
       "2  TARDBP        0        0        0        0        0        1        0   \n",
       "3  TARDBP        0        0        0        0        0        1        0   \n",
       "4  TARDBP        0        0        0        0        0        0        0   \n",
       "\n",
       "   NA19006  NA18948  ...  NA19058  NA18944  NA19084  NA19075  NA18950  \\\n",
       "0        0        0  ...        0        0        0        0        0   \n",
       "1        0        0  ...        0        0        0        0        0   \n",
       "2        0        0  ...        0        0        0        0        0   \n",
       "3        0        0  ...        0        0        0        0        0   \n",
       "4        0        0  ...        0        0        0        0        0   \n",
       "\n",
       "   NA19086  NA19085  NA19001  NA18954  NA18998  \n",
       "0        0        1        1        1        1  \n",
       "1        0        1        1        1        1  \n",
       "2        0        1        1        0        1  \n",
       "3        0        1        1        0        1  \n",
       "4        0        0        0        0        0  \n",
       "\n",
       "[5 rows x 106 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "jpt_infile = join(root_dir,\"populations/output/JPT/all.csv\")\n",
    "jpt_df=pd.read_csv(jpt_infile)\n",
    "jpt_df = jpt_df[[\"gene\"]+ jpt_genome_ids ]\n",
    "jpt_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d57add12-2a24-4031-a43a-b921e2e8ce6d",
   "metadata": {},
   "source": [
    "## T-test on Vietnamese (KHV) and Japanese (JPT) populations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "41b9b449-2cc9-4d34-b9f3-f7089ebf9c8d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mean</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ALS2</th>\n",
       "      <td>0.152535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANG</th>\n",
       "      <td>0.021324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATXN2</th>\n",
       "      <td>0.060198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C19ORF12</th>\n",
       "      <td>0.088971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C9ORF72</th>\n",
       "      <td>0.161412</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              mean\n",
       "gene              \n",
       "ALS2      0.152535\n",
       "ANG       0.021324\n",
       "ATXN2     0.060198\n",
       "C19ORF12  0.088971\n",
       "C9ORF72   0.161412"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_khv_df = khv_df.groupby([\"gene\"]).sum().sum(axis=1)\n",
    "g_khv_df = g_khv_df /len(khv_genome_ids)\n",
    "khv = pd.DataFrame()\n",
    "khv['mean']= g_khv_df / len(khv_genome_ids)\n",
    "khv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "a418fb87-6359-43bd-963a-2b699bf2a8bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mean</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ALS2</th>\n",
       "      <td>0.111655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANG</th>\n",
       "      <td>0.028118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATXN2</th>\n",
       "      <td>0.075556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C19ORF12</th>\n",
       "      <td>0.079728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C9ORF72</th>\n",
       "      <td>0.125533</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              mean\n",
       "gene              \n",
       "ALS2      0.111655\n",
       "ANG       0.028118\n",
       "ATXN2     0.075556\n",
       "C19ORF12  0.079728\n",
       "C9ORF72   0.125533"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_jpt_df = jpt_df.groupby([\"gene\"]).sum().sum(axis=1)\n",
    "g_jpt_df = g_jpt_df / len(jpt_genome_ids)\n",
    "jpt = pd.DataFrame()\n",
    "jpt['mean']= g_jpt_df / len(jpt_genome_ids)\n",
    "jpt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "dc4ac466-a672-4b80-9c71-79493fba0fc5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Two-Sample Independent T-Test:\n",
      "T-statistic: 0.1777\n",
      "P-value: 0.8595\n"
     ]
    }
   ],
   "source": [
    "t_statistic, p_value = stats.ttest_ind(khv['mean'], jpt['mean'], equal_var=True)\n",
    "print(f\"\\nTwo-Sample Independent T-Test:\")\n",
    "print(f\"T-statistic: {t_statistic:.4f}\")\n",
    "print(f\"P-value: {p_value:.4f}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "72e823ee-6dd6-4536-936a-43602ac148ef",
   "metadata": {},
   "source": [
    "## T-test on Vietnamese (KHV) and Finland (FIN) populations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "56e72d40-b63f-4b51-a72e-27a6e59d4530",
   "metadata": {},
   "outputs": [],
   "source": [
    "fin_genome_ids = ['HG00349', 'HG00327', 'HG00266', 'HG00323', 'HG00368', 'HG00326', 'HG00270', 'HG00321', 'HG00337', 'HG00189', 'HG00272', 'HG00382', 'HG00364', 'HG00304', 'HG00339', 'HG00330', 'HG00355', 'HG00186', 'HG00308', 'HG00353', 'HG00285', 'HG00273', 'HG00371', 'HG00274', 'HG00375', 'HG00282', 'HG00346', 'HG00345', 'HG00325', 'HG00367', 'HG00183', 'HG00366', 'HG00358', 'HG00275', 'HG00177', 'HG00178', 'HG00338', 'HG00319', 'HG00311', 'HG00381', 'HG00334', 'HG00277', 'HG00369', 'HG00324', 'HG00267', 'HG00372', 'HG00342', 'HG00284', 'HG00190', 'HG00181', 'HG00356', 'HG00180', 'HG00176', 'HG00271', 'HG00362', 'HG00288', 'HG00384', 'HG00182', 'HG00376', 'HG00336', 'HG00350', 'HG00318', 'HG00188', 'HG00344', 'HG00315', 'HG00269', 'HG00335', 'HG00179', 'HG00377', 'HG00185', 'HG00359', 'HG00383', 'HG00380', 'HG00276', 'HG00268', 'HG00332', 'HG00278', 'HG00365', 'HG00281', 'HG00343', 'HG00351', 'HG00187', 'HG00373', 'HG00320', 'HG00310', 'HG00171', 'HG00313', 'HG00357', 'HG00361', 'HG00379', 'HG00378', 'HG00309', 'HG00328', 'HG00290', 'HG00331', 'HG00329', 'HG00174', 'HG00312', 'HG00306', 'HG00341', 'HG00360', 'HG00280', 'HG00173']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "7c18b39e-f080-4e19-ac23-60b2ca1bb790",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>HG00349</th>\n",
       "      <th>HG00327</th>\n",
       "      <th>HG00266</th>\n",
       "      <th>HG00323</th>\n",
       "      <th>HG00368</th>\n",
       "      <th>HG00326</th>\n",
       "      <th>HG00270</th>\n",
       "      <th>HG00321</th>\n",
       "      <th>HG00337</th>\n",
       "      <th>...</th>\n",
       "      <th>HG00290</th>\n",
       "      <th>HG00331</th>\n",
       "      <th>HG00329</th>\n",
       "      <th>HG00174</th>\n",
       "      <th>HG00312</th>\n",
       "      <th>HG00306</th>\n",
       "      <th>HG00341</th>\n",
       "      <th>HG00360</th>\n",
       "      <th>HG00280</th>\n",
       "      <th>HG00173</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TARDBP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 104 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene  HG00349  HG00327  HG00266  HG00323  HG00368  HG00326  HG00270  \\\n",
       "0  TARDBP        0        0        0        0        1        0        1   \n",
       "1  TARDBP        0        0        0        0        0        0        0   \n",
       "2  TARDBP        0        0        0        0        0        0        0   \n",
       "3  TARDBP        0        0        0        0        0        0        1   \n",
       "4  TARDBP        0        0        0        0        0        0        0   \n",
       "\n",
       "   HG00321  HG00337  ...  HG00290  HG00331  HG00329  HG00174  HG00312  \\\n",
       "0        0        0  ...        0        1        0        0        0   \n",
       "1        0        0  ...        0        0        0        0        0   \n",
       "2        0        0  ...        0        0        0        0        0   \n",
       "3        0        0  ...        0        0        0        0        0   \n",
       "4        0        0  ...        0        0        0        0        0   \n",
       "\n",
       "   HG00306  HG00341  HG00360  HG00280  HG00173  \n",
       "0        0        0        0        0        1  \n",
       "1        0        0        0        0        0  \n",
       "2        0        0        0        0        0  \n",
       "3        0        0        1        0        0  \n",
       "4        0        0        0        0        0  \n",
       "\n",
       "[5 rows x 104 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fin_infile = join(root_dir,\"populations/output/FIN/all.csv\")\n",
    "fin_df=pd.read_csv(fin_infile)\n",
    "fin_df = fin_df[[\"gene\"]+ fin_genome_ids ]\n",
    "fin_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "95758649-6866-4dc6-b92a-2e8020ee88cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mean</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ALS2</th>\n",
       "      <td>0.086436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANG</th>\n",
       "      <td>0.047507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATXN2</th>\n",
       "      <td>0.126591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C19ORF12</th>\n",
       "      <td>0.090206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C9ORF72</th>\n",
       "      <td>0.198982</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              mean\n",
       "gene              \n",
       "ALS2      0.086436\n",
       "ANG       0.047507\n",
       "ATXN2     0.126591\n",
       "C19ORF12  0.090206\n",
       "C9ORF72   0.198982"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_fin_df = fin_df.groupby([\"gene\"]).sum().sum(axis=1)\n",
    "g_fin_df = g_fin_df /len(fin_genome_ids)\n",
    "fin = pd.DataFrame()\n",
    "fin['mean']= g_fin_df / len(fin_genome_ids)\n",
    "fin.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "4260d55d-7e9d-4827-af84-f7eaec8d6fec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Two-Sample Independent T-Test:\n",
      "T-statistic: -0.3998\n",
      "P-value: 0.6905\n"
     ]
    }
   ],
   "source": [
    "t_statistic, p_value = stats.ttest_ind(khv['mean'], fin['mean'], equal_var=True, nan_policy='omit')\n",
    "print(f\"\\nTwo-Sample Independent T-Test:\")\n",
    "print(f\"T-statistic: {t_statistic:.4f}\")\n",
    "print(f\"P-value: {p_value:.4f}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "265f2bfc-8531-4443-8af4-a67a8dfe4ec7",
   "metadata": {},
   "source": [
    "## Debug variant counts before using t-test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "0db197ab-9980-4569-b466-c45eddf1e8e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mean</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ALS2</th>\n",
       "      <td>0.152535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANG</th>\n",
       "      <td>0.021324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATXN2</th>\n",
       "      <td>0.060198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C19ORF12</th>\n",
       "      <td>0.088971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C9ORF72</th>\n",
       "      <td>0.161412</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              mean\n",
       "gene              \n",
       "ALS2      0.152535\n",
       "ANG       0.021324\n",
       "ATXN2     0.060198\n",
       "C19ORF12  0.088971\n",
       "C9ORF72   0.161412"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_khv_df = khv_df.groupby([\"gene\"]).sum().sum(axis=1)\n",
    "g_khv_df = g_khv_df /len(khv_genome_ids)\n",
    "khv = pd.DataFrame()\n",
    "khv['mean']= g_khv_df / len(khv_genome_ids)\n",
    "khv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "351eae6a-ded8-44f7-bd69-eae43a11f685",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ALS2</th>\n",
       "      <td>1495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANG</th>\n",
       "      <td>209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATXN2</th>\n",
       "      <td>590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C19ORF12</th>\n",
       "      <td>872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C9ORF72</th>\n",
       "      <td>1582</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          count\n",
       "gene           \n",
       "ALS2       1495\n",
       "ANG         209\n",
       "ATXN2       590\n",
       "C19ORF12    872\n",
       "C9ORF72    1582"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_khv_df = khv_df.groupby([\"gene\"]).sum().sum(axis=1)\n",
    "khv = pd.DataFrame()\n",
    "khv['count']= g_khv_df \n",
    "khv.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46519ec9-97b0-4930-a2bb-f2cc69d2c8cb",
   "metadata": {},
   "source": [
    "## For example, a list for KHV would have 99 numbers since there are 99 genomes. One number (representing a genome) would be the mean amount of variant counts across all genes for that genome. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "ed6bc7ea-7e85-4bad-a0dc-7485d1e66fcc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "HG02069     9.444444\n",
       "HG01846     9.979798\n",
       "HG02078    12.222222\n",
       "HG01851     9.919192\n",
       "HG01865     9.161616\n",
       "             ...    \n",
       "HG01841     8.757576\n",
       "HG01596     4.101010\n",
       "HG02141     8.282828\n",
       "HG02067    11.898990\n",
       "HG02057     7.737374\n",
       "Length: 99, dtype: float64"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_khv_df = khv_df.groupby([\"gene\"]).sum().sum() / len(khv_genome_ids)\n",
    "g_khv_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "e757d2ec-b54e-4bc8-89c4-cd87fe69c274",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[np.float64(9.444444444444445),\n",
       " np.float64(9.97979797979798),\n",
       " np.float64(12.222222222222221),\n",
       " np.float64(9.919191919191919),\n",
       " np.float64(9.16161616161616),\n",
       " np.float64(10.606060606060606),\n",
       " np.float64(12.565656565656566),\n",
       " np.float64(9.181818181818182),\n",
       " np.float64(11.11111111111111),\n",
       " np.float64(8.434343434343434)]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(g_khv_df.values)[0:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "49cc110b-bca6-40e2-8e95-367260f9237f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "HG00349     8.427184\n",
       "HG00327    11.553398\n",
       "HG00266    14.126214\n",
       "HG00323    13.669903\n",
       "HG00368    12.728155\n",
       "             ...    \n",
       "HG00306    12.466019\n",
       "HG00341    14.951456\n",
       "HG00360    15.019417\n",
       "HG00280    10.427184\n",
       "HG00173    11.699029\n",
       "Length: 103, dtype: float64"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_fin_df = fin_df.groupby([\"gene\"]).sum().sum() / len(fin_genome_ids)\n",
    "g_fin_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "f06684d5-ea90-4d6d-af54-5f011ced0f4e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NA18999     9.761905\n",
       "NA18960    10.342857\n",
       "NA18976    10.380952\n",
       "NA19081     9.028571\n",
       "NA18987     6.780952\n",
       "             ...    \n",
       "NA19086     5.266667\n",
       "NA19085     5.066667\n",
       "NA19001     8.828571\n",
       "NA18954     8.457143\n",
       "NA18998     7.952381\n",
       "Length: 105, dtype: float64"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_jpt_df = jpt_df.groupby([\"gene\"]).sum().sum() / len(jpt_genome_ids)\n",
    "g_jpt_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "8a705bce-8692-4988-829e-c8482e3362f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import tukey_hsd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "3aaacbe9-b577-4de4-a5c9-cd151ac1b921",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Column_A Column_B Column_C\n",
      "0        1    apple     True\n",
      "1        2   banana    False\n",
      "2        3   orange     None\n",
      "3      4.0      NaN      NaN\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Define your dictionary with lists of unequal lengths\n",
    "data = {\n",
    "    'Column_A': [1, 2, 3, 4],\n",
    "    'Column_B': ['apple', 'banana', 'orange'],\n",
    "    'Column_C': [True, False]\n",
    "}\n",
    "\n",
    "# Create the DataFrame directly from the dictionary\n",
    "df = pd.DataFrame.from_dict(data, orient='index')\n",
    "df = df.transpose()\n",
    "# Print the resulting DataFrame\n",
    "print(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "105da27d-f606-496a-9865-5505f3aeeb17",
   "metadata": {},
   "outputs": [],
   "source": [
    "tdf = pd.DataFrame.from_dict({\n",
    "    \"KHV\": list(g_khv_df.values),\n",
    "    \"FIN\": list(g_fin_df.values),\n",
    "    \"JPT\": list(g_jpt_df.values)\n",
    "}, orient='index')\n",
    "tdf = tdf.transpose()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "6e9a38b4-36c0-4cac-8120-426a777a46c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>KHV</th>\n",
       "      <th>FIN</th>\n",
       "      <th>JPT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9.444444</td>\n",
       "      <td>8.427184</td>\n",
       "      <td>9.761905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9.979798</td>\n",
       "      <td>11.553398</td>\n",
       "      <td>10.342857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12.222222</td>\n",
       "      <td>14.126214</td>\n",
       "      <td>10.380952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9.919192</td>\n",
       "      <td>13.669903</td>\n",
       "      <td>9.028571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9.161616</td>\n",
       "      <td>12.728155</td>\n",
       "      <td>6.780952</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         KHV        FIN        JPT\n",
       "0   9.444444   8.427184   9.761905\n",
       "1   9.979798  11.553398  10.342857\n",
       "2  12.222222  14.126214  10.380952\n",
       "3   9.919192  13.669903   9.028571\n",
       "4   9.161616  12.728155   6.780952"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tdf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "08beb7e7-d72d-4c12-a15e-4ee6e77e727d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>KHV</th>\n",
       "      <th>FIN</th>\n",
       "      <th>JPT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>4.101010</td>\n",
       "      <td>15.708738</td>\n",
       "      <td>5.942857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>8.282828</td>\n",
       "      <td>14.300971</td>\n",
       "      <td>5.466667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>11.898990</td>\n",
       "      <td>9.854369</td>\n",
       "      <td>8.619048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>7.737374</td>\n",
       "      <td>12.466019</td>\n",
       "      <td>6.866667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>NaN</td>\n",
       "      <td>14.951456</td>\n",
       "      <td>6.238095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>NaN</td>\n",
       "      <td>15.019417</td>\n",
       "      <td>5.266667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>NaN</td>\n",
       "      <td>10.427184</td>\n",
       "      <td>5.066667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>NaN</td>\n",
       "      <td>11.699029</td>\n",
       "      <td>8.828571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.457143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.952381</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           KHV        FIN       JPT\n",
       "95    4.101010  15.708738  5.942857\n",
       "96    8.282828  14.300971  5.466667\n",
       "97   11.898990   9.854369  8.619048\n",
       "98    7.737374  12.466019  6.866667\n",
       "99         NaN  14.951456  6.238095\n",
       "100        NaN  15.019417  5.266667\n",
       "101        NaN  10.427184  5.066667\n",
       "102        NaN  11.699029  8.828571\n",
       "103        NaN        NaN  8.457143\n",
       "104        NaN        NaN  7.952381"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tdf.tail(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "9d045fb5-1ff0-4953-b048-33671a7c5bca",
   "metadata": {},
   "outputs": [],
   "source": [
    "result = tukey_hsd(g_khv_df.values, g_jpt_df.values, g_fin_df.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "478e7ddc-a061-4bc2-9393-d38777720536",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[1.00000000e+00, 1.28306179e-03, 3.33066907e-16],\n",
       "       [1.28306179e-03, 1.00000000e+00, 3.33066907e-16],\n",
       "       [3.33066907e-16, 3.33066907e-16, 1.00000000e+00]])"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result.pvalue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "74a3fc63-8450-4bff-a44a-c4118e143ab6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tukey's HSD Pairwise Group Comparisons (95.0% Confidence Interval)\n",
      "Comparison  Statistic  p-value  Lower CI  Upper CI\n",
      " (0 - 1)      1.040     0.001     0.351     1.730\n",
      " (0 - 2)     -3.880     0.000    -4.573    -3.187\n",
      " (1 - 0)     -1.040     0.001    -1.730    -0.351\n",
      " (1 - 2)     -4.920     0.000    -5.603    -4.237\n",
      " (2 - 0)      3.880     0.000     3.187     4.573\n",
      " (2 - 1)      4.920     0.000     4.237     5.603\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61e48911-aba0-4f9e-8eba-e16b1654a6df",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:insightgen]",
   "language": "python",
   "name": "conda-env-insightgen-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
